src/Schema/Keywords/Type.php

```diff
@@ -38,56 +37,40 @@ class Type extends BaseKeyword
     public function validate($data, string $type, ?string $format = null) : void
     {
         try {
-            Validator::in([
-                0 => 'boolean',
-                1 => 'object',
-                2 => 'array',
-                3 => 'number',
-                4 => 'integer',
-                5 => 'string',
-                // Note that there is no null type; instead, the nullable attribute is used as a modifier of the base type.
-            ])->stringType()->assert($type);
-
-            if ($this->parentSchema->nullable && $data === null) {
-                return;
-            }
-
             switch ($type) {
-                case 'boolean':
+                case CebeType::BOOLEAN:
                     if (! is_bool($data)) {
-                        throw new Exception(sprintf("Value '%s' is not a boolean", $data));
+                        throw TypeException::becauseTypeDoesNotMatch(CebeType::BOOLEAN, $data);
                     }
                     break;
-                case 'object':
+                case CebeType::OBJECT:
                     if (! is_object($data) && ! is_array($data)) {
-                        throw new Exception(sprintf("Value '%s' is not an object", $data));
+                        throw TypeException::becauseTypeDoesNotMatch(CebeType::OBJECT, $data);
```

Comment:
I like how it looks, `throw X::becauseY`. Will steal it :)

Reply:
Yeah, we using in our team. Looks nice, but not in 100% cases